tests/suites/test_suite_ssl.function

```diff
@@ -6180,3 +6189,77 @@ exit:
     USE_PSA_DONE( );
 }
 /* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED */
+void ssl_ecjpake_set_password( int use_opaque_arg,
+                               int err_injection_step_arg,
+                               int expected_error_arg )
+{
+    mbedtls_ssl_context ssl;
+    mbedtls_ssl_config conf;
+#if defined( MBEDTLS_USE_PSA_CRYPTO )
+    mbedtls_svc_key_id_t pwd_slot = MBEDTLS_SVC_KEY_ID_INIT;
+#else   /* MBEDTLS_USE_PSA_CRYPTO */
+    (void) use_opaque_arg;
+#endif  /* MBEDTLS_USE_PSA_CRYPTO */
+    const unsigned char pwd_string[ sizeof(ECJPAKE_TEST_PWD) ] = "";
+    size_t pwd_len = 0;
+    ecjpake_err_inj_step_t err_injection_step = err_injection_step_arg;
+    int ret;
+
+    USE_PSA_INIT( );
+
+    mbedtls_ssl_init( &ssl );
+
+    if( err_injection_step == ECJPAKE_ERR_UNITIALIZED_SSL_CONTEXT )
```

Comment:
This is not really error injection, just checking different behaviours.
To be honest, the function flow is a bit hard to follow. You can treat this as a minor for now, as this is subjective, but I would like to suggest a different approach:
How about following the similar pattern used in other function checks, and make this function a "sanity test" function, that tries to do things that error out starting from earliest steps? If the `mbedtls_ssl_set_hs_ecjpake_password` and `_opaque` calls are not destructive, the flow would be as follows:
Initialize things (up to `mbedtls_ssl_init`), then, call 
`mbedtls_ssl_set_hs_ecjpake_password` that errors out as expected, call `mbedtls_ssl_set_hs_ecjpake_password_opaque` if PSA is on that errors out as expected, **but then continue:**
do the `mbedtls_ssl_config_init` and next calls up to `mbedtls_ssl_setup`, call the `set_xxx_password` functions again and verify errors, then finally handle passwords and add final, working calls that would return successes. 
If `mbedtls_ssl_set_hs_ecjpake_password_opaque` is fixed and it doesn't destroy the external (from its point of view) key, I think that it should work, and there shouldn't be any parameters needed in this test function, as it would do everything itself.
Also the flow would be a bit cleaner as it would be linear, with errors next to calls.